packages/flutter/lib/src/foundation/constants.dart

```diff
@@ -68,10 +68,5 @@ const bool kDebugMode = !kReleaseMode && !kProfileMode;
 const double precisionErrorTolerance = 1e-10;
 
 /// A constant that is true if the application was compiled to run on the web.
-///
-/// This implementation takes advantage of the fact that JavaScript does not
-/// support integers. In this environment, Dart's doubles and ints are
-/// backed by the same kind of object. Thus a double `0.0` is identical
-/// to an integer `0`. This is not true for Dart code running in AOT or on the
-/// VM.
-const bool kIsWeb = identical(0, 0.0);
+/// ignore: unnecessary_const
+const bool kIsWeb = !const bool.fromEnvironment('dart.library.io');
```

Comment:
Should be the same as?

```suggestion
const bool kIsWeb = !bool.fromEnvironment('dart.library.io');
```